### ISSUE TITLE
Sarif report missing endColumn/endLine

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#4919]
## Steps to Reproduce

Make any rule fail which highlights specific parts of the code (e.g. class name).

## Observed Behavior
<details><summary>txt</summary>

```
UndocumentedPublicProperty - [updater] at P:\projects\workspace\net.twisterrob.sun\component\widget\src\main\java\net\twisterrob\sun\android\SunAngleWidgetProvider.kt:14:23 - Signature=SunAngleWidgetProvider.kt$SunAngleWidgetProvider$@Inject lateinit var updater: SunAngleWidgetUpdater
```
</details>

<details><summary>html</summary>

```html
  <details id="UndocumentedPublicProperty" open="open">
    <summary class="rule-container"><span class="rule">UndocumentedPublicProperty: 9 </span><span class="description">Public properties require documentation.</span></summary>
    <ul>
      <li><span class="location">component/widget/src/main/java/net/twisterrob/sun/android/SunAngleWidgetProvider.kt:14:23</span><span class="message">The property updater is missing documentation.</span>
        <pre><code><span class="lineno">  11 </span>
<span class="lineno">  12 </span>class SunAngleWidgetProvider : LoggingAppWidgetProvider() {
<span class="lineno">  13 </span>
<span class="lineno">  14 </span>	@Inject lateinit var <span class="error">updater</span>: SunAngleWidgetUpdater
<span class="lineno">  15 </span>	@Inject lateinit var locations: LocationRetriever
<span class="lineno">  16 </span>
<span class="lineno">  17 </span>	override fun onReceive(context: Context, intent: Intent) {
</code></pre>
      </li>
```
![image](https://user-images.githubusercontent.com/2906988/172064997-62db86b7-50fe-4b9a-a79b-94e42e1f3f45.png)

</details>

<details><summary>xml</summary>

```xml
<file name="component/widget/src/main/java/net/twisterrob/sun/android/SunAngleWidgetProvider.kt">
	<error line="14" column="23" severity="warning" message="The property updater is missing documentation." source="detekt.UndocumentedPublicProperty" />
</file>
```

</details>

<details><summary>sarif</summary>

```json
        {
          "level": "warning",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "component/widget/src/main/java/net/twisterrob/sun/android/SunAngleWidgetProvider.kt",
                  "uriBaseId": "%SRCROOT%"
                },
                "region": {
                  "startColumn": 23,
                  "startLine": 14
                }
              }
            }
          ],
          "message": {
            "text": "The property updater is missing documentation."
          },
          "ruleId": "detekt.comments.UndocumentedPublicProperty"
        },
```

![image](https://user-images.githubusercontent.com/2906988/172064813-f1fd272e-d9c6-4145-9079-7517a963f4ad.png)

</details>

## Expected Behavior

- HTML is good as it highlights the correct part already. One enhancement could be to output the range of failure. Although visual highlight is there already.
- TXT is missing the output range, it would be important here as there's no visual representation.
- XML is Checkstyle, so nothing to do as their [model](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java) doesn't support it.
- SARIF has an extremely flexible model, it very likely supports everything as Detekt's internal model and way more. endColumn and endLine exists, so it should be utilized and immediate effect in DX improvement can be observed when GitHub Sarif uploader is used.

## Context
I have recently set up Detekt on GitHub on my codebases and started seeing failures. In many cases the Detekt HTML report highlights the issue more clearly than SARIF annotations on GitHub. This locks in the use of an un-mergeable format which is per-module/per-variant, so processing is impossible and therefore each module's report has to be examined independently. SARIF nicely fixes this issue but its support is usually lacking (not just Detekt, lint too) because it's a very complex schema.

## Your Environment

* Version of detekt used: 1.20.0
* Version of Gradle used (if applicable): 7.4.2
* Gradle scan link (add `--scan` option when running the gradle task): https://scans.gradle.com/s/chtvgcipi5tpk
* Operating System and version: N/A
* Link to your project (if it's a public repository): https://github.com/TWiStErRob/net.twisterrob.sun/pull/95 see failure and annotations on [4aad14e](https://github.com/TWiStErRob/net.twisterrob.sun/pull/95/commits/4aad14e1187988b0308712f35fa1e840aef6e9d8)

**Comment by BraisGabin** [comment]
I think that the only one that should be fixed here is sarif. We are going to remove `txt` as it is a nearly useless report and now that we have markdown support it is even more useless.

**Comment by VitalyVPinchuk** [comment]
Hello
I'd like to work on this

**Comment by BraisGabin** [comment]
Go for it
